### PR TITLE
Doc for deviated PoroElastic wellbore

### PR DESCRIPTION
**Pull request by sytuannguyen**
This PR do:
- Add rst, xml and validation script for the example of injecting in a deviated poroelastic open hole
- Add rst, xml and validation script for the example of drilling a deviated poroelastic wellbore in a pre-stressed and pre-pressurized formation  
- Minor change to optimize the figures in the docs for some wellbore problems
- Change the name of the poromechanic example to vertical poro-elasto-plastic for a better description
- Remove two redundant wellbore xmls that are not linked to any doc or integrated test.
- Remove two redundant python scripts related to wellbore problems

### CODE REVIEW

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+Deviated Poro-Elastic Wellbore Problem
+####################################################
+
+------------------------------------------------------------------
+Problem description
+------------------------------------------------------------------
+
+This example aims to solve a typical injection problem of a deviated poroelastic open wellbore subjected to a fluid pressure injected inside the hole. The problem geometry is generated by the interal wellbore mesh generator. The poroelastic coupled solver, that is a combination of the solid mechanic solver and the single phase flow solver, is employed to solver this coupling problem. In-situ stress and pressure are excluded from this example.
+
+Analytical results of the fluid pressure, and the radial and hoop stresses around the wellbore are expressed in the Laplace space as `(Detournay and Cheng, 1988) <https://www.sciencedirect.com/science/article/abs/pii/0148906288922991>`__ :
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst:14]:
```suggestion
Analytical solutions of the pore pressure, the radial and hoop stresses in the near wellbore region are expressed in the Laplace space as `(Detournay and Cheng, 1988) <https://www.sciencedirect.com/science/article/abs/pii/0148906288922991>`__ :
```

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+.. math::
+    p = p_0 \frac{kn(0,R \sqrt{s})}{s kn(0,\sqrt{s})}
+
+.. math::
+    \sigma_{rr} = -b \frac{1-2\nu}{1-\nu} p_0 \frac{-R * kn(1,R \sqrt{s}) + kn(1,\sqrt{s})}{R^2 \sqrt{s^3} kn(0,\sqrt{s})}
+
+.. math::
+    \sigma_{\theta\theta} = -b \frac{1-2\nu}{1-\nu} p - \sigma_{rr}
+
+where :math:`s` is the Laplace variable, :math:`R` is the dimensionless radial coordinate that is defined by the radial coordinate normalized by the hole radius, :math:`\nu` the Poisson ratio and :math:`b` the Biot coefficient. Fluid pressure and stresses in time space are obtained from these analytical expression by the inverse Laplace transformation (see the attached Python script for more details).
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst:25]:
```suggestion
where :math:`s` is the Laplace variable, :math:`kn` is the is the modified Bessel function of second kind of order n, :math:`R` is the dimensionless radial coordinate that is defined by the radial coordinate normalized by the wellbore radius, :math:`\nu` is the Poisson ratio and :math:`b` is the Biot coefficient. Fluid pressure and stresses in time space are obtained from these analytical expressions by inversing Laplace transformation (see the attached Python script for more details).  
```

Comment (jhuang2601) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst:25]: Please verify the analytical solution. Here, `s` should be an expression of Laplace variable and diffusion coefficient. This analytical solutions are expressed in the local coordinate system (along wellbore trajectory). I suggest to make a note here.

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+  src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml
+
+
+In this example, we would focus our attention on the ``Solver`` and the ``Mesh`` tags.
+
+-----------------------------------------------------------
+Poroelastic solver
+-----------------------------------------------------------
+
+The coupled ``Poroelastic`` solver, that is a coupling between the solid mechanic solver ``SolidMechanicsLagrangianSSLE`` and the single phase flow solver ``SinglePhaseFVM``, is required for solving this wellbore problem.
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst:44]:
```suggestion
The coupled ``Poroelastic`` solver, that defines a coupling strategy between the solid mechanics solver ``SolidMechanicsLagrangianSSLE`` and the single phase flow solver ``SinglePhaseFVM``, is required for solving this wellbore problem.  
```

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+  :language: xml
+  :start-after: <!-- SPHINX_WellboreMesh -->
+  :end-before: <!-- SPHINX_WellboreMeshEnd -->
+
+
+--------------------------------------------------------------------   
+Constitutive law
+--------------------------------------------------------------------
+
+Isotropic ``PoroElasticIsotropic`` constitutive behavior, with the specified bulk and shear elastic moduli and the Biot coefficient, is considered for the rock around the wellbore. Fluid constitutive properties such as dynamic viscosity, compressibility, etc. are given in the ``CompressibleSinglePhaseFluid`` constitutive block.
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst:80]:
```suggestion
Isotropic constitutive behavior ``PoroElasticIsotropic`` , with the specified bulk and shear elastic moduli, is considered for the rock around the wellbore. Fluid properties, such as dynamic viscosity and compressibility, are given in the ``CompressibleSinglePhaseFluid`` constitutive block. 
```

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+  :language: xml
+  :start-after: <!-- SPHINX_Material -->
+  :end-before: <!-- SPHINX_MaterialEnd -->
+
+
+--------------------------------------------------------------------	       
+Boundary conditions
+--------------------------------------------------------------------
+
+Far-field boundaries are subjected to roller constraints and impermeable condition. The pressure on the wellbore wall is defined by face pressure field specification. The nodeset generated by the internal wellbore generator for this face is named as ``rneg``. The negative sign of the scale is attributed to the negative sign convention for compressive stresses in GEOSX. Initial fluid pressure, reference porosity and permeability are also given for the whole domain. In this example, uniform isotropic permeability is assumed.
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst:92]:
```suggestion
Far-field boundaries are impermeable and subjected to roller constraints. The pressure on the wellbore wall is defined by face pressure field specification. The nodeset generated by the internal wellbore generator for this face is named as ``rneg``. The negative sign of the scale denotes the fluid injection. Initial fluid pressure, reference porosity and permeability are also given for the computational domain. In this example, uniform isotropic permeability is assumed.
```

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+.. literalinclude:: ../../../../../../src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml
+  :language: xml
+  :start-after: <!-- SPHINX_BoundaryConditions -->
+  :end-before: <!-- SPHINX_BoundaryConditionsEnd -->
+
+---------------------------------
+Results and benchmark
+---------------------------------
+
+Result of the fluid pressure distribution at an injection time of 78 (s) is shown in the figure below:
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst:103]:
```suggestion
Result of the fluid pressure distribution after 78 s injection is shown in the figure below:
```

File: src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml [hunk 7]
```diff
@@ -5,6 +5,7 @@
   <Solvers
```
Comment (jhuang2601) [inline src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml:5]: Please prepare a coarse version of this example for integrated test.

**Review by jhuang2601 — COMMENTED** [review]
Good job!

**Comment by jhuang2601** [comment]
@sytuannguyen Please add the xmls of your wellbore problems (casedWell, deviatedElastic and deviatedPoroElastic) to the corresponding integratedTest folder. @shabnamjs or @joshua-white can help to add them to the integrated tests.

File: src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml [hunk 8]
```diff
@@ -129,7 +138,8 @@
       defaultBulkModulus="11039657020.4"
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml]: These values are computed from E, nu, bBiot, MBiot taken from a reference of Abousleiman et al.

File: src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml [hunk 9]
```diff
@@ -140,7 +150,7 @@
 
     <BiotPorosity
       name="rockPorosity"
-      grainBulkModulus="1.0e27"
+      grainBulkModulus="48208109259"
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml]: see comment above

File: src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml [hunk 10]
```diff
@@ -129,7 +138,8 @@
       defaultBulkModulus="11039657020.4"
       defaultShearModulus="8662741799.83"/>
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml]: see comment above

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst (hunk 1 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst]: Done

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst (hunk 2 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst]: Done

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst (hunk 3 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst]: Done

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst (hunk 4 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst]: Done

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst (hunk 5 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst]: Done

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst (hunk 6 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst]: Done

File: src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml (hunk 7 above)
Comment (sytuannguyen) [reply src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml]: Done

**Comment by sytuannguyen** [comment]
> @sytuannguyen Please add the xmls of your wellbore problems (casedWell, deviatedElastic and deviatedPoroElastic) to the corresponding integratedTest folder. @shabnamjs or @joshua-white can help to add them to the integrated tests.

Done for this deviatedPoroElastic case. For the other cases, I will not touch to avoid conflicts because the xml files are being moving to the new location.

**Review by jhuang2601 — APPROVED** [review]

**Comment by sytuannguyen** [comment]
@jhuang2601 I'm going to add also a doc for the case with in-situ stresses.

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+.. literalinclude:: ../../../../../../inputFiles/multiphysics/DeviatedPoroElasticWellbore_Drilling_Base.xml
+  :language: xml
+  :start-after: <!-- SPHINX_InsituConditions -->
+  :end-before: <!-- SPHINX_InsituConditionsEnd -->
+
+---------------------------------
+Results and benchmark
+---------------------------------
+
+Example of the pore pressure distribution after 78 s injection is shown in the figure below:
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst:40]:
```suggestion
Pore pressure distribution after 78 s injection is shown in the figure below:
```

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst [hunk 12]
```diff
@@ -0,0 +1,326 @@
...
+
+.. figure:: pressure_drilling.png
+   :align: center
+   :width: 800
+   :figclass: align-center
+
+
+A good agreement between the GEOSX results and the corresponding analytical solutions is shown in the figure below:
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst:48]:
```suggestion
A good agreement between the GEOSX results and the corresponding analytical solutions `(Abousleiman and Cui, 1998) <https://www.sciencedirect.com/science/article/pii/S0020768398001012>`__ is shown in the figure below:
```

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst [hunk 13]
```diff
@@ -0,0 +1,326 @@
+.. _AdvancedExampleDeviatedPoroElasticWellboreDrilling:
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst:1]: This example is a fluid depletion problem with far field stresses; whereas, the DeviatedPoroElasticWellbore is a fluid injection problem without in-situ stresses, right? Is there any other difference between the two examples?

Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst:1]: @jhuang2601 Yes, the you are right. The first example mimic pore pressure injection problem without considering in-situ stresses and in-situ pore pressure. The second example mimic the drilling problem of a deviated wellbore in a pre-stressed and pressurized basin that leads to complex loading combinations including anisotropic compression (as considered by Detournay and Cheng 1988) and longitudinal shearing (as added by Abousleiman and Cui 1998). 

May be I will add elastic anisotropy to the second example. Analytical results exist also for the transversely isotropic case.

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst [hunk 14]
```diff
@@ -0,0 +1,326 @@
...
+
+
+#########################################################################################
+Drilling a deviated Poro-Elastic Wellbore
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst]: done

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst [hunk 15]
```diff
@@ -0,0 +1,326 @@
...
+#########################################################################################
+
+------------------------------------------------------------------
+Problem description
+------------------------------------------------------------------
+
+This example deal with the problem of drilling a deviated poro-elastic wellbore. This can be regarded as an extension of the poroelastic wellbore example :ref:`AdvancedExampleDeviatedPoroElasticWellbore` with consideration of in-situ stresses and pore pressure effects. Both pore pressure and mud pressure are supposed to be nil at the borehole wall. The in-situ horizontal stress is anisotropic. The wellbore trajectory is deviated from the direction of the main in-situ stresses. Analytical solutions of the pore pressure, the radial and hoop stresses in the near wellbore region are expressed in the Laplace space as `(Abousleiman and Cui, 1998) <https://www.sciencedirect.com/science/article/pii/S0020768398001012>`__.
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst]: done

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst (hunk 11 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst]: done

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst (hunk 12 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst]: done

File: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst [hunk 16]
```diff
@@ -0,0 +1,326 @@
...
+
+  inputFiles/multiphysics/DeviatedPoroElasticWellbore_Drilling_Base.xml
+
+
+.. code-block:: console
+
+  inputFiles/multiphysics/DeviatedPoroElasticWellbore_Drilling_Example.xml
+
+
+They are almost idential to those of the example :ref:`AdvancedExampleDeviatedPoroElasticWellbore` except the ``FieldSpecifications`` tag. Indeed, for the boundary conditions, we need consider following additional field specifications to defined the in-situ stresses and pore pressure as well as the zero pore pressure at the borehole wall.
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example2.rst]: done

**Review by jhuang2601 — APPROVED** [review]

**Comment by sytuannguyen** [comment]
@jhuang2601 I think the "smoke" xmls ca be added to the list of integrated tests after the PR that reorgenizes the XML will be merged. For the moment, I think we can just keep them out of the list such that we can remove the rebaseline requirement that I cannot do myself.

**Review by rrsettgast — APPROVED** [review]

**Comment by klevzoff** [comment]
@sytuannguyen This PR has a conflict now, could you take a look?

**Comment by sytuannguyen** [comment]
@klevzoff the conflict has been resolved and the PR has passed the Travis builds

**Pull request merged**